doc/quotas.md

```diff
@@ -0,0 +1,110 @@
+# Quotas
+
+In large Graphite (go-carbon) storage installations, it's desirable to have control how many resources a user can consume.
+
+Go-Carbon addresses this issue by the implementation pattern-matching based quota design.
+
+Limitations/Caveats:
+
+* The current implementation is based on concurrent/realtime trie-index, so to use quotas, also need to enable those features on go-carbon.
+* Pattern matching rules aren't regexp, but like graphite query syntax, using globs
+* Quota and usage metrics are only produced if
+
+## Available controls
+
+* namespaces: how many sub-namespaces are allowed for the matched prefix/namespaces
+* metrics: how many metrics are allowed for the matched prefix/namespaces
+* logical-size: how many logical disk space are allowed for the matched prefix/namespaces
+* physical-size: how many physical disk space are allowed for the matched prefix/namespaces
+* data-points: how many data points (inferred using storage-schemas.conf) are allowed for the matched prefix/namespaces
```

Comment:
@bom-d-van Do we really need this kind of fine-graining? As above, which use-cases does this cover?

Reply:
there are more explanation bellow about data-points and logical-size, there are some overlaps between two of them.
like other control options listed there, could be enabled if needed.